protocol/src/did_webvh/service/resolver/resolver_service.rs

```diff
@@ -108,164 +103,159 @@ fn verify_proofs<C: std::error::Error>(
-            .map_err(|_| {
+        let current_version_time =
+            DateTime::parse_from_rfc3339(&log_entry.version_time).map_err(|_| {
                 DidWebvhResolverError::ResolveIdentifier("Failed to parse version time".to_string())
             })?;
         if current_version_time > current_time {
             return Err(DidWebvhResolverError::ResolveIdentifier(
                 "Log entry time is in the future".to_string(),
             ));
         }
-        // check existence of update_keys in the first log entry, with check is_empty
-        let mut update_keys = match first_log_entry.parameters.update_keys.as_deref() {
-            None | Some(&[]) => {
-                return Err(DidWebvhResolverError::ResolveIdentifier(
-                    "No update keys found".to_string(),
-                ))
-            }
-            Some(update_keys) => update_keys,
-        };
-
-        verify_proofs(update_keys, first_log_entry)?;
-
-        // verify entry hash
-        // recalc scid and entry hash
-        let mut replaced_hash_entry = first_log_entry.replace_to_placeholder()?;
-        let calculated_scid = replaced_hash_entry.calc_entry_hash()?;
-        replaced_hash_entry.replace_placeholder_to_id(&calculated_scid)?;
-
-        let calculated_hash = replaced_hash_entry.calc_entry_hash()?;
-        if calculated_hash != hash {
+        let previous_version_time = DateTime::parse_from_rfc3339(&previous_entry.version_time)
+            .map_err(|_| {
+                DidWebvhResolverError::ResolveIdentifier("Failed to parse version time".to_string())
+            })?;
+        if current_version_time > previous_version_time {
             return Err(DidWebvhResolverError::ResolveIdentifier(
-                format!(
-                    "Entry hash is not matched: calc {} != {}",
-                    calculated_hash, hash
-                )
-                .to_string(),
+                "Log entry time is in the future".to_string(),
             ));
         }
 
-        let mut previous_version_number = id;
-        let mut previous_entry = first_log_entry.clone();
-
-        // verify the log entry iter start next entry
-        for log_entry in rest_log_entries.iter() {
-            // check version number is sequential
-            let (id, hash) = log_entry.parse_version_id()?;
-            if id != previous_version_number + 1 {
-                return Err(DidWebvhResolverError::ResolveIdentifier(
-                    "Log entry version is not sequential".to_string(),
-                ));
-            }
+        // verify vertion_id, current log entry's version_id is generated from previous log
+        // entry's version_id
+        let (_, previous_hash) = previous_entry.parse_version_id()?;
+        let mut tmp_entry = log_entry.clone();
+        tmp_entry.version_id = previous_hash;
+        let recalculated_hash = tmp_entry.calc_entry_hash()?;
```

Comment:
細かいですが、ここはブロック式にしてスコープが小さくしてもよさそうですね

```suggestion
        let recalculated_hash = {
                let (_, previous_hash) = previous_entry.parse_version_id()?;
                let mut tmp_entry = log_entry.clone();
                tmp_entry.version_id = previous_hash;
                tmp_entry.calc_entry_hash()?;
        };

```